src/main/java/org/apache/commons/collections4/bloomfilter/IndexProducer.java

```diff
@@ -125,9 +121,7 @@ class Indices {
             private int size;
 
             boolean add(final int index) {
-                if (size == data.length) {
-                    data = Arrays.copyOf(data, (int) Math.min(MAX_ARRAY_SIZE, size * 2L));
-                }
+                data = IndexUtils.ensureCapacityForAdd(data, index);
```

Comment:
This is wrong. It should use `size` when checking the capacity:
```java
data = IndexUtils.ensureCapacityForAdd(data, size);
```
I put this into `IndexProducerTest`:
```java
@ParameterizedTest
@ValueSource(ints = {32, 33})
void testAsIndexArray(int n) {
    IndexProducer ip = i -> {
        for (int j = 0; j < n; j++) {
            // Always test index zero
            i.test(0);
        }
        return true;
    };
    Assertions.assertArrayEquals(new int[n], ip.asIndexArray());
}
```
The size 33 fails as the initial array of capacity 32 is not enlarged.